src/main/java/me/rigamortis/seppuku/impl/module/movement/PullDownModule.java

```diff
@@ -0,0 +1,51 @@
+package me.rigamortis.seppuku.impl.module.movement;
+
+import me.rigamortis.seppuku.api.event.EventStageable;
+import me.rigamortis.seppuku.api.event.player.EventPlayerUpdate;
+import me.rigamortis.seppuku.api.module.Module;
+import me.rigamortis.seppuku.api.value.NumberValue;
+import net.minecraft.client.Minecraft;
+import team.stiff.pomelo.impl.annotated.handler.annotation.Listener;
+
+/**
+ * @author cookiedragon234
+ * (cookiedragon234 owns me and all)
+ * also buy backdoored client kk thx
+ */
+public class PullDownModule extends Module
+{
+	public PullDownModule()
+	{
+		super("PullDown", new String[]{"FastFall"}, "Increase your downwards velocity when falling", "NONE", -1, ModuleType.MOVEMENT);
+	}
+	
+	public final NumberValue<Float> speed = new NumberValue<>("Speed", new String[]{"velocity"}, 10f, Float.class, 0f, 20f, 1f);
+	
+	@Listener
+	public void onUpdate(EventPlayerUpdate event)
+	{
+		if (event.getStage() == EventStageable.EventStage.PRE)
+		{
+			final Minecraft mc = Minecraft.getMinecraft();
+			
+			// obvs dont do this when flying or when using elytras
+			if(mc.player.isElytraFlying() || mc.player.capabilities.isFlying || mc.player.onGround) return;
+			
+			// dont trigger when they could just be jumping, 3 blocks is maybe overkill? But its probably the best thing to do
+			boolean isBlockBelow =
```

Comment:
Why don't you just trace a few blocks downwards and check if you get a valid result?  What if there's water there, why can't we just shoot into it instead of slowing?  What if we're going to die, do we really want to shoot downwards extremely fast?

Reply:
Downwards water velocity is checked due to it being swimming rather than falling (I think? Cant remember though).

Nothing I can do about fall damage, that would happen anyway. Also fall damage isnt affected by velocity but by distance, so pull down wouldnt make a difference in that situation.

I can add the raytracing.

Reply:
Pull down would make a major difference as the time for human thinking may have been cut into an eighth of what it could have been.  This means less time for you to think about any possible solution to save yourself.

Perhaps you could make a way for the user to confirm them falling, maybe with certain modes.  One mode forcefully throws them downwards while the other mode requires a certain key to be held.  Or even simpler, one mode that checks for a key to be held, if that key is bound to nothing, then throw them downwards.

Reply:
Done

Reply:
And what I mean through water being below us is, why slow down for those 3 blocks if we have a very high chance of living anyways?  I also want to emphasize that we trace the path of the bounding box for the player, this will give us the best idea of if we land inside the water.

Reply:
I dont understand. You seem to be combining the fall damage and water issues. Fall damage is resolved. I dont understand why water is an issue. This is for accelerating air fall speed not water fall speed.

Reply:
Okay, we slow down 3 blocks above the ground for what purpose?  Why slow down and delay us moving from the area faster?  Efficiency is key, if we can expedite some process, we'll do it.  What I'm trying to say is that we should just not slow down those 3 blocks above the water.  We'll be safe if we hit the water, just keep going and stop applying the velocity modifier when we hit the water.  

The only thing I can see this being used for is to be faster than someone else.  Why not make it smart enough to be the fastest?